File: src/Entity/Renewal.php
<?php

namespace App\Entity;

use App\Repository\RenewalRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: RenewalRepository::class)]
class Renewal
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    private ?\DateTimeInterface $previousDueDate = null;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    private ?\DateTimeInterface $newDueDate = null;

    #[ORM\Column(type: Types::DATETIME_MUTABLE)]
    private ?\DateTimeInterface $renewedAt = null;

    #[ORM\Column]
    private int $renewalNumber = 0;

    #[ORM\ManyToOne(inversedBy: 'renewals')]
    #[ORM\JoinColumn(nullable: false)]
    private ?Loan $loan = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getPreviousDueDate(): ?\DateTimeInterface
    {
        return $this->previousDueDate;
    }

    public function setPreviousDueDate(\DateTimeInterface $previousDueDate): static
    {
        $this->previousDueDate = $previousDueDate;
        return $this;
    }

    public function getNewDueDate(): ?\DateTimeInterface
    {
        return $this->newDueDate;
    }

    public function setNewDueDate(\DateTimeInterface $newDueDate): static
    {
        $this->newDueDate = $newDueDate;
        return $this;
    }

    public function getRenewedAt(): ?\DateTimeInterface
    {
        return $this->renewedAt;
    }

    public function setRenewedAt(\DateTimeInterface $renewedAt): static
    {
        $this->renewedAt = $renewedAt;
        return $this;
    }

    public function getRenewalNumber(): int
    {
        return $this->renewalNumber;
    }

    public function setRenewalNumber(int $renewalNumber): static
    {
        $this->renewalNumber = $renewalNumber;
        return $this;
    }

    public function getLoan(): ?Loan
    {
        return $this->loan;
    }

    public function setLoan(?Loan $loan): static
    {
        $this->loan = $loan;
        return $this;
    }
}

File: libraryhub-loan-desktop/src/main/java/model/Renewal.java
package model;

import java.time.LocalDate;
import java.time.LocalDateTime;

public class Renewal {
    private Integer id;
    private LocalDate previousDueDate;
    private LocalDate newDueDate;
    private LocalDateTime renewedAt = LocalDateTime.now();
    private int renewalNumber;
    private Integer loanId;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public LocalDate getPreviousDueDate() {
        return previousDueDate;
    }

    public void setPreviousDueDate(LocalDate previousDueDate) {
        this.previousDueDate = previousDueDate;
    }

    public LocalDate getNewDueDate() {
        return newDueDate;
    }

    public void setNewDueDate(LocalDate newDueDate) {
        this.newDueDate = newDueDate;
    }

    public LocalDateTime getRenewedAt() {
        return renewedAt;
    }

    public void setRenewedAt(LocalDateTime renewedAt) {
        this.renewedAt = renewedAt;
    }

    public int getRenewalNumber() {
        return renewalNumber;
    }

    public void setRenewalNumber(int renewalNumber) {
        this.renewalNumber = renewalNumber;
    }

    public Integer getLoanId() {
        return loanId;
    }

    public void setLoanId(Integer loanId) {
        this.loanId = loanId;
    }

    @Override
    public String toString() {
        return "Renewal #" + (id == null ? "new" : id);
    }
}
